src/main/scala/cromwell/binding/WdlExpression.scala

```diff
@@ -188,8 +189,26 @@ object WdlExpression {
             }
           case _ => Failure(new WdlExpressionException("Right-hand side of expression must be identifier"))
         }
+      case a: Ast if a.isArrayOrMapIndex =>
+        val index = evaluate(a.getAttribute("rhs"), lookup, functions, interpolateStrings)
+        val mapOrArray = evaluate(a.getAttribute("lhs"), lookup, functions, interpolateStrings)
+        (mapOrArray, index) match {
+          case (Success(a: WdlArray), Success(i: WdlInteger)) =>
+            Try(a.value(i.value)) match {
+              case s:Success[WdlValue] => s
+              case Failure(ex) => Failure(new WdlExpressionException(s"Failed to find index $index on array:\n\n$mapOrArray\n\n${ex.getMessage}"))
```

Comment:
Could you use the constructor for exception which takes the 'caused by' exception as an argument? e.g. New WdlExpressionException("...", ex)